File: python/src/main/python/stores/YogurtIceCreamStore.py
from icecream.yogurt.GuavaYogurtIceCream import *
from icecream.yogurt.MangoYogurtIceCream import *
from icecream.yogurt.PineappleYogurtIceCream import *
from icecream.yogurt.StrawberryYogurtIceCream import *
from IceCreamStore import *


class YogurtIceCreamStore(IceCreamStore):
    def createIceCream(self, flavor):
        if (flavor == "Strawberry"):
            return StrawberryYogurtIceCream()
        elif (flavor == "Pineapple"):
            return PineappleYogurtIceCream()
        elif (flavor == "Mango"):
            return MangoYogurtIceCream()
        elif (flavor == "Guava"):
            return GuavaYogurtIceCream()
        return None

File: csharp/factory-method/YogurtIceCreamStore.cs
using factorymethod.main.icecreams;
using factorymethod.main.icecreams.yogurt;

namespace factorymethod.main.stores
{
    internal class YogurtIceCreamStore : IceCreamStore
    {
        override
            protected IceCream createIceCream(string flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryYogurtIceCream();
            }

            if (flavor.Equals("Pineapple"))
            {
                return new PineappleYogurtIceCream();
            }
            if (flavor.Equals("Mango"))
            {
                return new MangoYogurtIceCream();
            }
            return flavor.Equals("Guava") ? new GuavaYogurtIceCream() : null;
        }

    }
}
}